x/storage/keeper/keeper.go

```diff
@@ -886,6 +886,10 @@ func (k Keeper) DeleteObject(
 		return types.ErrSourceTypeMismatch
 	}
 
+	if objectInfo.ObjectStatus == types.OBJECT_STATUS_CREATED {
+		return k.CancelCreateObject(ctx, operator, bucketName, objectName, types.CancelCreateObjectOptions{SourceType: types.SOURCE_TYPE_ORIGIN})
+	}
+
 	if objectInfo.ObjectStatus != types.OBJECT_STATUS_SEALED &&
```

Comment:
This code should also be refactored. Object has three status, so this condition is equal to objectInfo.ObjectStatus != types.OBJECT_STATUS_CREATED